packages/block-editor/src/components/rich-text/index.js

```diff
@@ -289,6 +299,24 @@ export function RichTextWrapper(
 		anchorRef.current?.focus();
 	}
 
+	const bindings = getBlockAttributes( clientId )?.metadata?.bindings;
```

Comment:
You should subscribe to these selector inside a `useSelect` callback. When the attributes change, the component will currently not re-render.

Reply:
Do you mean something like this?

```js
const bindings = useSelect(
	( select ) => {
		return select( blockEditorStore ).getBlockAttributes( clientId )
			?.metadata?.bindings;
	},
	[ clientId ]
);
```

Reply:
It also looks like there is some room for improvements to avoid computations in the case when there are no bindings set.

Reply:
I've moved most of the logic to the initial `useSelect` callback that returns earlier if the block isn't selected: [link](https://github.com/WordPress/gutenberg/pull/58085/commits/7c1ca5a2bf0a1fa8a23bc56938ad2e6eddaa5c58). Let me know if that's what you had in mind.

> It also looks like there is some room for improvements to avoid computations in the case when there are no bindings set.

Regarding this, the only optimization I see after the last modification is to only get the `getBlockBindingsSource` function when bindings exist ([link](https://github.com/WordPress/gutenberg/blob/add/block-bindings-support-in-the-editor/packages/block-editor/src/components/rich-text/index.js#L136)) with something like this:

```
const { getBlockBindingsSource } = blockBindings
			? unlock( select( blockEditorStore ) )
			: {};
```

Would that make sense? Which other improvements do you see?